### ISSUE TITLE
crash when splitting lines

### ISSUE THREAD

**Issue by ocehugo** [jupyter/jupyter_console#115]
Info:
Jupyter 4.2.1
Python 3.5.2
ipython 5.2.0

The following is ocurring sometimes when you search something in the history and try to modify the command to run again. I believe it's something related to the "new way" that console now is accepting inputs/using readline.

```python
In [415]:
Traceback (most recent call last):
  File "/usr/local/bin/jupyter-console", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.5/site-packages/jupyter_core/application.py", line 267, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/traitlets/config/application.py", line 658, in launch_instance
    app.start()
  File "/usr/local/lib/python3.5/site-packages/jupyter_console/app.py", line 151, in start
    self.shell.mainloop()
  File "/usr/local/lib/python3.5/site-packages/jupyter_console/ptshell.py", line 463, in mainloop
    self.interact()
  File "/usr/local/lib/python3.5/site-packages/jupyter_console/ptshell.py", line 448, in interact
    code = self.prompt_for_code()
  File "/usr/local/lib/python3.5/site-packages/jupyter_console/ptshell.py", line 408, in prompt_for_code
    reset_current_buffer=True)
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/interface.py", line 394, in run
    self.eventloop.run(self.input, self.create_eventloop_callbacks())
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/eventloop/posix.py", line 164, in run
    t()
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/eventloop/posix.py", line 83, in read_from_stdin
    inputstream.feed(data)
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/terminal/vt100_input.py", line 395, in feed
    self._input_parser.send(c)
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/terminal/vt100_input.py", line 304, in _input_parser_generator
    self._call_handler(match, prefix)
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/terminal/vt100_input.py", line 337, in _call_handler
    self.feed_key_callback(KeyPress(key, insert_text))
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/interface.py", line 1023, in feed_key
    cli.input_processor.process_keys()
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/key_binding/input_processor.py", line 201, in process_keys
    self._process_coroutine.send(key_press)
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/key_binding/input_processor.py", line 158, in _process
    self._call_handler(matches[-1], key_sequence=buffer)
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/key_binding/input_processor.py", line 228, in _call_handler
    handler.call(event)
  File "/usr/local/lib/python3.5/site-packages/prompt_toolkit/key_binding/registry.py", line 31, in call
    return self.handler(event)
  File "/usr/local/lib/python3.5/site-packages/jupyter_console/ptshell.py", line 342, in _
    more, indent = self.check_complete(d.text)
  File "/usr/local/lib/python3.5/site-packages/jupyter_console/ptshell.py", line 426, in check_complete
    more = (code.splitlines()[-1] != "")
IndexError: list index out of range
```

I can't find a way to reproduce this but it's happening quite frequent ( I had to downgrade to avoid crashing my sessions). 

It's appears that the list is missing an update/grab the text or maybe it's just a try/except missing around...

**Comment by takluyver** [comment]
Thanks, I think this was already fixed by #95, but I guess we may need to make another release of jupyter_console with that fix.

**Comment by takluyver** [comment]
5.1 is out now
